### ISSUE TITLE
🐛 BUG: A component used outside of `<body>` or `<head>` in a layout breaks compiled output

### ISSUE THREAD

**Issue by FredKSchott** [snowpackjs/astro#1038]
### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug

From Discord: https://discord.com/channels/830184174198718474/845451724738265138/872977023214186537
/cc @brycewray 

The bug is that the actual output of our compiler isn't valid JS, probably due to some assumption that components outside of `<head>` break. This only seems to be effecting layouts that contain the entire `<html>` element. I couldn't reproduce on a page without a layout.

@natemoo-re this is probably best solved in your refactoring work. Not a huge priority, but whenever you work on layout components it would be good to solve for this and add a test

### Steps to Reproduce

1. Clone the repo
2. see the error
3. add `<head>` around the `<Head>` component in the layout file
4. see the error resolved!

### Link to Minimal Reproducible Example (Optional)

https://github.com/brycewray/astro-site

**Comment by brycewray** [comment]
I will leave the linked repo as-is for now, but have confirmed locally that this worked for me. Thanks again, @FredKSchott!

**Comment by natemoo-re** [comment]
Confirming that layout handling has had a complete overhaul in the `astro@next` preview! This should no longer be a problem.

Will put this in our milestone to be closed when that release is stable.

**Comment by natemoo-re** [comment]
Following up on my previous comment, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
